### DISCUSSION TITLE
CI build failed

### DISCUSSION THREAD

**Discussion by deep-soft** [rustdesk/rustdesk#3858] [General]
Tried to find discussion about this failed build using GitHub actions ...

All jobs from Actions, which depend on vcpkg, failed (starting yesterday).

In this step:
      - name: Restore from cache and install vcpkg
        uses: lukka/run-vcpkg@v7
        with:
          setupOnly: true
          vcpkgGitCommitId: ${{ env.VCPKG_COMMIT_ID }}

tried to change
uses: lukka/run-vcpkg@v7
with
uses: lukka/run-vcpkg@v10

but not working.
Need change here: github.com/microsoft/vcpkg
There is one pull request #30546 on vcpkg, but is waiting for approval
Reference: https://github.com/microsoft/vcpkg/pull/30546

**Comment by rustdesk** [comment]
@Kingtous

**Comment by Kingtous** [comment]
Thanks for finding the problem, maybe we need can build prebuilt so for windows x64 and x86? just like the Android and arm,arm64 so build in https://github.com/Kingtous/rustdesk_thirdpary_lib.

**Reply by deep-soft** [reply]
Could be a temporary solution until pull request approved on microsoft/vcpkg
And any speedup is good (and needed)!

**Reply by Kingtous** [reply]
Now the PR is here now. :)

@rustdesk https://github.com/rustdesk/rustdesk/pull/3861

**Reply by deep-soft** [reply]
Wow, that was quick... trying now ...

**Comment by deep-soft** [comment]
LE:
pull request approved and applied:
https://github.com/microsoft/vcpkg/commit/69efe9cc2df0015f0bb2d37d55acde4a75c9a25b

**Comment by deep-soft** [comment]
vcpkg fixed... for now.

Compilation for Windows (flutter, x64) possible two ways:
# 1  -----------------------------------------------------------------
env:
  # vcpkg 2023.04.01 - msys2-runtime-3.4.6-1
  VCPKG_COMMIT_ID: "69efe9cc2df0015f0bb2d37d55acde4a75c9a25b"

# ...

      - name: Restore from cache and install vcpkg
        uses: lukka/run-vcpkg@v10
        with:
          vcpkgGitCommitId: ${{ env.VCPKG_COMMIT_ID }}

      - name: Install vcpkg dependencies
        run: |
          $VCPKG_ROOT/vcpkg install libvpx:x64-windows-static libyuv:x64-windows-static opus:x64-windows-static
        shell: bash

      - name: Build rustdesk
        run: python3 .\build.py --portable --hwcodec --flutter --feature IddDriver

# or:
# 2  -----------------------------------------------------------------
      - name: Install vcpkg dependencies
        run: |
          cd C:\
          git clone https://github.com/Kingtous/rustdesk_thirdpary_lib --depth=1

      - name: Build rustdesk
        env:
          VCPKG_ROOT: C:\rustdesk_thirdpary_lib\vcpkg
        run: python3 .\build.py --portable --hwcodec --flutter --feature IddDriver
